changes/3715.fixed

```diff
@@ -0,0 +1,4 @@
+Fixed inability to set the `failover_strategy` field as unspecified/blank when creating/updating `DeviceRedundancyGroup` via the REST API.
+Fixed incorrect interim natural key definition for `Device` and `VRF` models.
+Fixed the possibility of setting values for nonexistent custom fields via the REST API.
+Fixed incorrect password-handling logic when creating/updating a `User` via the REST API.
```

Comment:
Does this resolve #3438

Reply:
Good catch! I think it does. Mind taking a look at the tests I've added in `users/test_api.py` and let me know if anything is missing there?

Reply:
I verified in your branch that:
- Able to create a user with `is_superuser` set to True or False, and `is_staff` set to True or False
- Able to modify an existing user password
This should resolve #3438

Also #2208 should be resolved by this PR because:
- GET http://localhost:8080/api/users/users/{id}/ returns `is_superuser` and `last_login` fields

Reply:
3500 gets the credit for resolving #2208 by switching the UserSerializer fields to `__all__`.  :-) Thanks for checking these!